src/DynamoCore/Graph/Nodes/PortModel.cs

```diff
@@ -550,6 +558,14 @@ internal string GetOutPortType()
 
             return null;
         }
+
+        public override void Dispose()
+        {
+            if (HasBeenDisposed) return;
```

Comment:
Is HasBeenDisposed being set to true in the base Dispose() method? If not, you need to set it to true at the end of this function.